### DISCUSSION TITLE
Dummy data on third party graphql query?

### DISCUSSION THREAD

**Discussion by Blakeinstein** [gatsbyjs/gatsby#27614] [Help] [answered]
Using `gatsby-source-github-api` I implemented getting data from github on my gatsby project.

This has raised two issues,

- My tests on pull-requests will fail as github secrets are not available on Pull Requests, as such I cannot provide my token for testing.
- Anyone trying to contribute to the project, will have issues as their api key wont have necessary privileges.

Is there a way I can provide dummy data for testing?

**Comment by herecydev** [comment, marked as answer]
I've had great success with mocking the `useStaticQuery` to return dummy data. If you use page Queries you can pass in a data prop with the dummy data as well. This works well for a simple schema, if you have a lot of custom resolvers the complexity can become somewhat problematic but none-the-less usable.

There's some great info on this page: https://www.gatsbyjs.com/docs/unit-testing/#mocking-gatsby
